### DISCUSSION TITLE
problem redirecting to the home assistant address, always 400: Bad Request

### DISCUSSION THREAD

**Discussion by JR-aaas** [hassio-addons/addon-nginx-proxy-manager#246] [Q&A] [answered]
I have a newly installed home assistant, set up according to the instructions and everything worked until last week.
 page 400: Bad Request

![image](https://user-images.githubusercontent.com/62484901/125678962-2d490671-71f7-4dbb-953d-14c55aefbe2a.png)

Setup is correct 

![image](https://user-images.githubusercontent.com/62484901/125679130-59df550b-0ac6-4197-b248-c2204c405ab6.png)

HTTP website is correct  

![image](https://user-images.githubusercontent.com/62484901/125679316-c1f023b7-5349-4a79-801c-ab5612a9ce95.png)

HTTPS

![image](https://user-images.githubusercontent.com/62484901/125679397-bad83cf2-2f02-495e-9a25-670136f98283.png)

![image](https://user-images.githubusercontent.com/62484901/125680218-a45ebfad-ed64-492e-bb5c-d968f584c76f.png)

I don't know what to do, please advise why this is happening?

**Comment by frenck** [comment, marked as answer]
You haven't set up Home Assistant correctly for use with reverse proxies. Please consult the Home Assistant documentation for that.

https://www.home-assistant.io/integrations/http#use_x_forwarded_for

**Reply by JR-aaas** [reply]
Thank you very much for the quick reply.
Thank you for the direction.

**Reply by subw0ofer** [reply]
I have added following to my configuration.yaml and still receiving error 400, what am I doing wrong?

http:
  use_x_forwarded_for: true
  trusted_proxies:
    - 192.168.55.85/32  # Add the IP address of the proxy server -> IP of HassOS-Host

**Reply by subw0ofer** [reply]
Found the solution myself now: 
Just use the default configuration without changing IP works.

http:
  use_x_forwarded_for: true
  trusted_proxies:
    - 172.30.33.0/24  # Add the IP address of the proxy server

**Reply by Dudleydogg** [reply]
I would like to ask a question the first part of the config says to Forward ports 80 and 443 to home assistant but does not specify what ports on homeassistant to forward too?  my home assistant runs on 8123 could you finish  that statement, thanks

**Reply by Dudleydogg** [reply]
Would also ask for DNS challenge to include HE.NET which has a fully automated option

**Reply by monghuz** [reply]
I'd like to add that the revers proxy's IP address is not necessarily the same as the one you are accessing it's web UI (details below).

In my case NGINX Proxy Manager (_192.168.x.y:81_) was installed within HomeAssistant (_192.168.x.y:8123_) which was running in a VM.
I added _192.168.x.y/32_ as a trusted proxy, but still got HTTP400. 
At this point I found the below warnings in `/config/home-assistant.log`.
`ERROR (MainThread) [homeassistant.components.http.forwarded] Received X-Forwarded-For header from an untrusted proxy 172.30.33.3`
The IP above is a IP of NPMs container and they are communicating with each other via this docker defined network, hence I had to add `172.30.33.3` to make it work.

At the end I setup NPM outside of HomeAssistant os my other services won't get impacted If I would break :)

**Reply by Dark1886** [reply]
@frenck I know this is an old thread, but this is exactly what I am seeing on my system, but I can't seem to get the problem fixed. |

My current issue: https://github.com/hassio-addons/addon-nginx-proxy-manager/issues/471#event-10582941983

**Reply by T1bolus** [reply]
> I'd like to add that the revers proxy's IP address is not necessarily the same as the one you are accessing it's web UI (details below).
> 
> In my case NGINX Proxy Manager (_192.168.x.y:81_) was installed within HomeAssistant (_192.168.x.y:8123_) which was running in a VM. I added _192.168.x.y/32_ as a trusted proxy, but still got HTTP400. At this point I found the below warnings in `/config/home-assistant.log`. `ERROR (MainThread) [homeassistant.components.http.forwarded] Received X-Forwarded-For header from an untrusted proxy 172.30.33.3` The IP above is a IP of NPMs container and they are communicating with each other via this docker defined network, hence I had to add `172.30.33.3` to make it work.
> 
> At the end I setup NPM outside of HomeAssistant os my other services won't get impacted If I would break :)

Thank you so much. You helped me alot. Cannot understand why it is that complicated to get a reverse proxy running :)


I would also add, that you should NOT use a fixed ip like: 172.30.33.3
If youre adding or removing some addons (or container in reality) this ip can chanhge. So better add: 172.30.33.0/32

**Comment by no1s3e** [comment]
Hi, I got an error when I use the nginx addon, there were no such errors before the upgrade to 2021.7. Registered in configuration.yaml http: use_x_forwarded_for: true trusted_proxies: - 192.168.1.0/24 - 172.30.33.0/24 - 172.30.32.0/24 the error does not disappear.
![Screenshot_2021-07-22-00-41-42-308_io homeassistant companion android](https://user-images.githubusercontent.com/70854652/126564393-4d56eacd-0d83-4642-988d-a18b5825c278.jpg)

**Reply by schlumm** [reply]
I wanted to get into the HA using duckDNS from outside of my home. After installing the nginx add-on. I cannot access to the HA main webpage. using the ip address with port 8183 or 8123, says site cannot be reached. Trying restart--> no help.
Using port 81, I get into the nginx and added multiple proxy hosts but cannot access to HA (even to an arduino server, that I can get to it directly using its IP address).
What am I missing?
How can I get to the configuration YAML and also to the nginx configuration (to verify that I've put everything OK), if I cannot get to HA?
